src/ossia-max/src/utils.cpp

```diff
@@ -376,64 +343,118 @@ ossia::value atom2value(t_symbol* s, int argc, t_atom* argv)
-  ossia::max::model* model = nullptr;
+  static int count = 0;
+  count++;
+  std::vector<object_base*> objects_to_register;
 
-  int start_level = 0;
-  int view_level = 0;
-  int model_level = 0;
+  ossia_max::instance().patchers[patcher].loadbanged = true;
 
-  if (x->m_otype == object_class::view)
-    start_level = 1;
+  t_object* root_patcher{};
+  if(caller)
+    root_patcher = caller->m_patcher;
 
-  view = find_parent_box_alive<ossia::max::view>(x, start_level, &view_level);
+  // 1: look for device, client, model and view objects into the patcher
+  auto& pat_desc = ossia_max::instance().patchers[patcher];
 
-  if (x->m_otype == object_class::model
-      || x->m_otype == object_class::remote)
+  if(root_patcher != patcher)
   {
-    model =  find_parent_box_alive<ossia::max::model>(x, start_level, &model_level);
+    device_base* db = pat_desc.device?static_cast<device_base*>(pat_desc.device):
+                                      static_cast<device_base*>(pat_desc.client);
+    if(db && db != caller)
+    {
+      if(db->m_device)
+      {
+        db->m_registered = true;
+        return register_children_in_patcher_recursively(patcher, db);
+      }
+      else
+        return;
+    }
   }
 
-  t_object* object = nullptr;
-  ossia::max::view* tmp = nullptr;
+  node_base* nb = pat_desc.model?static_cast<node_base*>(pat_desc.model):
+                                 static_cast<node_base*>(pat_desc.view);
 
-  // FIXME this will fail as soon as https://github.com/ossia/libossia/issues/208 is implemented
-  // or if model and view are mixed in the same hierarchy
-
-  while (view)
+  if(nb && nb != caller)
   {
-    vs.push_back(view->m_name->s_name);
-    tmp = view;
-    view = find_parent_box_alive<ossia::max::view>(tmp, 1, &view_level);
+    nb->update_path();
+    switch(nb->m_otype)
+    {
+      case object_class::model:
+      {
+        auto mdl = static_cast<model*>(nb);
+        mdl->do_registration();
+        register_children_in_patcher_recursively(patcher, mdl);
+        break;
+      }
+      case object_class::view:
+      {
+        auto vw = static_cast<view*>(nb);
+        vw->do_registration();
+        register_children_in_patcher_recursively(patcher, vw);
+        break;
+      }
+      default:
+        break;
+    }
+    return;
   }
 
-  ossia::max::model* tmp_model = nullptr;
-  while (model)
+  register_objects_by_type(pat_desc.parameters);
+  register_objects_by_type(pat_desc.remotes);
+  register_objects_by_type(pat_desc.attributes);
+
+  for(auto subpatcher : pat_desc.subpatchers)
   {
-    vs.push_back(model->m_name->s_name);
-    tmp_model = model;
-    model
-        = find_parent_box_alive<ossia::max::model>(tmp_model, 1, &model_level);
+    register_children_in_patcher_recursively(subpatcher, caller);
   }
+  count--;
 
-  fmt::memory_buffer fullpath;
-  auto rit = vs.rbegin();
-  for (; rit != vs.rend(); ++rit)
+  // when the first call end, fire all values (when there is some)
+  if(count = 0)
```

Comment:
== ? and the code is a bit weird, the if does nothing (well, except assigning 0 to count which does not feel like a good thing :p

Reply:
I think I passed the Ballmer peak